### PR TITLE
Check BIP148 on historical blocks and rewind if necessary

### PR DESCRIPTION
**Pull request by luke-jr**

### CODE REVIEW

File: src/validation.cpp [hunk 1]
```diff
@@ -1852,17 +1852,8 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
...
-         (nMedianTimePast <= 1510704000) &&  // Wed 15 Nov 2017 00:00:00 UTC
-         (!IsWitnessLockedIn(pindex->pprev, chainparams.GetConsensus()) &&  // Segwit is not locked in
-          !IsWitnessEnabled(pindex->pprev, chainparams.GetConsensus())) )   // and is not active.
-    {
-        bool fVersionBits = (pindex->nVersion & VERSIONBITS_TOP_MASK) == VERSIONBITS_TOP_BITS;
-        bool fSegbit = (pindex->nVersion & VersionBitsMask(chainparams.GetConsensus(), Consensus::DEPLOYMENT_SEGWIT)) != 0;
-        if (!(fVersionBits && fSegbit)) {
-            return state.DoS(0, error("ConnectBlock(): relayed block must signal for segwit, please upgrade"), REJECT_INVALID, "bad-no-segwit");
-        }
+    if (!CheckBIP148(pindex, chainparams.GetConsensus())) {
```
Comment (NicolasDorier) [inline src/validation.cpp]: need checking  `bool fCheckBIP148 = GetBoolArg("-bip148", DEFAULT_BIP148);`

Comment (luke-jr) [reply src/validation.cpp]: This fork doesn't have BIP148 optional.

**Comment by achow101** [comment]
utACK c68f7724c4cd3b237b49a2efa6dfc3454e0a2f36

File: src/validation.cpp (hunk 1 above)
Comment (NicolasDorier) [reply src/validation.cpp]: ah ok you updated it.  You were using it in RewindBlockIndex.

**Comment by luke-jr** [comment]
Rebased

**Comment by mkwia** [comment]
Does this account for a user that might have BIP148 enabled to begin with, but then turn it off?

**Comment by luke-jr** [comment]
No, it is a one-way transition to BIP148.

**Pull request merged**